File: src/c/_15_3Sum.c
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>
#include <string.h>
#include <math.h>
#include <limits.h>
#include <stdbool.h>

int **threeSum(int *nums, int numsSize, int *returnSize, int **returnColumnSizes)
{
    int **returnColumnSize = 3;
    for (int i = 0; i < numsSize; i++)
    {
        for (int j = i + 1; j < numsSize; j++)
        {
            for (int k = j + 1; k < numsSize; k++)
            {
                if (nums[i] + nums[j] + nums[k] == 0)
                {
                    int *temp = (int *)malloc(3 * sizeof(int));
                    temp[0] = i;
                    temp[1] = j;
                    temp[2] = k;
                }
            }
        }
    }
}

// Driver code
int main(int argc, char const *argv[])
{
    /* code */
    return 0;
}

File: src/python/_15_3Sum.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        ans = []
        nums.sort()
        size = len(nums)
        for i, n in enumerate(nums):
            if i > 0 and n == nums[i - 1]:
                continue
            l = i + 1
            r = size - 1
            while l < r:
                value = nums[l] + nums[r]
                if value < -n:
                    l += 1
                elif value > -n:
                    r -= 1
                else:
                    ans.append([n, nums[l], nums[r]])
                    l += 1
                    while l < r and nums[l] == nums[l - 1]:
                        l += 1
        return ans
